src/kernel.rs

```diff
@@ -52,34 +43,31 @@ impl EraInfo {
         self.era_id
     }
 
+    #[allow(clippy::result_large_err)]
     pub fn validate(
         &self,
-        block_header_with_signatures: &BlockHeaderWithSignatures,
+        JsonBlockWithSignatures { block, proofs }: JsonBlockWithSignatures,
```

Comment:
@uchu Does this work with what the new side-care outputs when you try to get block by height?